### PR TITLE
chore: adds JSON validation example.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR moves https://github.com/tetratelabs/wasm-json-validation-demo into this repo to be provided as a first class example.

Ping @deva26 @liamawhite @M4tteoP

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,10 +2,15 @@ module github.com/tetratelabs/proxy-wasm-go-sdk
 
 go 1.17
 
-require github.com/stretchr/testify v1.7.0
+require (
+	github.com/stretchr/testify v1.7.0
+	github.com/tidwall/gjson v1.14.0
```
Comment (jcchavezs) [inline go.mod:7]: I think at least this example should be an own go module to avoid adding unrelated deps to this module. How about that?

Comment (mathetake) [reply go.mod:7]: oh yeah agreed. it's time to have separate gomod for each example. Would you mind doing that in another PR?

File: examples/json_validation/main.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ type pluginConfiguration struct {
 // Override types.DefaultPluginContext.
 func (ctx *pluginContext) OnPluginStart(pluginConfigurationSize int) types.OnPluginStartStatus {
 	data, err := proxywasm.GetPluginConfiguration()
-	if err != nil {
+	if err != nil && err != types.ErrorStatusNotFound {
```
Comment (jcchavezs) [inline examples/json_validation/main.go:48]: cc @mathetake I think this is common enough that maybe shouldn't be an error? I don't know.

**Review by deva26 — APPROVED** [review]
LGTM. Thanks.

File: examples/json_validation/main_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		expectedAction types.Action
+	}
+
+	for name, tCase := range map[string]testCase{
+		"pauses due to invalid payload": {
+			body:           "invalid_payload",
+			expectedAction: types.ActionPause,
+		},
+		"pauses due to unknown keys": {
+			body:           "{\"unknown_key\":\"unknown_value\"}",
```
Comment (morepork) [inline examples/json_validation/main_test.go:58]: These json bodies are easier to read using raw strings as you don't need to escape
```suggestion
			body:           `{"unknown_key":"unknown_value"}`,
```

**Review by morepork — APPROVED** [review]

File: examples/json_validation/main.go (hunk 2 above)
Comment (mathetake) [reply examples/json_validation/main.go:49]: this is lined with Proxy-Wasm's system call specification where not found is defined as error, not Go specific https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/exports.cc#L475-L480

Comment (mathetake) [reply examples/json_validation/main.go:49]: but yeah we could change and just return empty slice instead

Comment (mathetake) [reply examples/json_validation/main.go:49]: I mean, internally

File: go.mod (hunk 1 above)
Comment (jcchavezs) [reply go.mod]: Done: https://github.com/tetratelabs/proxy-wasm-go-sdk/pull/265

**Comment by mathetake** [comment]
Build failure seems due to the working directory for `tinygo build`? https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/main/Makefile#L10

**Review by mathetake — APPROVED** [review]
Thanks @jcchavezs!

**Pull request merged**